### DISCUSSION TITLE
mrv2 vs tlRender

### DISCUSSION THREAD

**Discussion by JohanAberg** [ggarra13/mrv2#247] [General]
Hello!
I've been evaluating both mrv2 and tlRender and I'm impresssed with the functionality and performance so far!
From what I understand, mrv2 uses some of tlRenders code.

Can you please clarify the relationship between the two projects, just out of curiosity.

Cheers

**Comment by ggarra13** [comment]
Sure!  I'll be glad to!

tlRender was born as a library for video and audio playback written mainly by Darby Johnston.  For testing, he developed two work-in-progress players called tlplay-qt (based on Qt5/Qt6) and later tlplay (based on GLFW).  tlplay is mainly Darby's baby that he is planning to release it in the near future.  It is small and mean and is perfect for playing on platforms where OpenGL is not available but GLES is (Tablets, Phones, etc).

mrv2 (which is based on FLTK 1.4) is a video program that uses a fork of tlRender which gets updated every couple of weeks with the latest from Darby and I also add my own code to it. When mrv2 introduces big forks in tlRender I make a push request for Darby to merge my changes into tlRender (sometimes he does, sometimes he doesn't).  Compared to tlplay, mrv2 supports more codecs (VP9, dav1d, Cineform, etc) for both loading and saving, supports annotations, PDF exporting, proper color video encoding, Python, has Wayland support, supports hardware encoding on macOS, etc.  However, it does not run on Tablets or Phones as of yet.

Oh yes... I forgot.  Darby's source code is **VERY** clean, while mrv2's source code is still more messy.

Hope that clears it!

**Reply by JohanAberg** [reply]
This is a great overview! Thank you!
